### ISSUE TITLE
Update tsconfig.json schema (`files` property is missing, potentially other props as well)

### ISSUE THREAD

**Issue by Meligy** [Microsoft/TypeScript#8138]
Hello,
Can you please update `tsconfig.json` schema with support for the `files` property and possible other missing properties?

Thanks a lot.

**P.S.**

I understand that there is a version of TypeScript tsconfig.json schema maintained by the Atom Typescript 3rd party package, There is also a [version](http://json.schemastore.org/tsconfig) that's also linked [in official docs](https://www.typescriptlang.org/docs/handbook/tsconfig.json.html#schema).

This is why I posted this here, as I'm not sure where this relation is at. Sorry if this is the wrong place.

**Comment by mhegazy** [comment]
@Meligy i am not sure i understand the issue. i see the "files" property in http://json.schemastore.org/tsconfig:

```js
  "definitions": {
    "filesDefinition": {
      "properties": {
        "files": {
          "type": "array",
          "description": "If no 'files' property is present in a tsconfig.json, the compiler defaults to including all files the containing directory and subdirectories. When a 'files' property is specified, only those files are included.",
          "items": {
            "type": "string",
            "format": "uri"
          }
        }
      },
      "not": {
        "required": [ "exclude" ]
      }
    },
```

**Comment by Meligy** [comment]
Hmm. Looks like I acted blindly on this. I got an error in VS Code for the `files` property. I thought maybe their schema is outdated, and replaced the contents of the files with this one. It still did not respect the property. It shows it in autocomplete, but when I add it, the editor says the file is not following the schema.

I assumed it's the schema file itself not being updated. It could be something else. I'll close this one for now. Thanks for looking at it :S
